Rocket.Chat/Extensions/UIViewControllerExtension.swift

```diff
@@ -7,10 +7,59 @@
 //
 
 import UIKit
+import Photos
+import MobileCoreServices
 import ObjectiveC
 
 private var viewScrollViewAssociatedKey: UInt8 = 0
 
+protocol MediaPicker: AnyObject { }
```

Comment:
@RocketChat/ios here I would have to replicate the exact same code for launching the camera and photo library than we already had on `ChatViewController`, then I thought that it would be nice to add it into an extension. Firstly I thought it would be cool to put it into `UIViewController` extension, but it  doesn't make sense to have these media picking related code for all view controllers.
Assuming that we can't add constraints to `UIViewController` extensions I thought that the best way to emulate this behavior was to create a protocol (MediaPicker) and constrain it only to view controllers that implement the delegates needed to provide the media picking behavior.

It works for camera/photo library as for drawing, even though we only have the drawing code in one place at our codebase it will avoid us to replicate code in the future if we have to add drawing to another view controller.

To be honest I'm in doubt if this is an elegant solution or a tricky one 🤔 so please share your thoughts.

Reply:
I like it, feels really nice :+1:

Reply:
@filipealva I think this is perfect for now, in case we need to have something more generic in the future we can change it. :-)

Reply:
Cool @filipealva !

We could be doing this for our "Alerter" protocol also:

```swift
protocol Alerter: class {
    func alert(title: String, message: String, handler: ((UIAlertAction) -> Void)?)
}

extension Alerter where Self: UIViewController { // doing extension UIViewController: Alerter instead
    func alert(title: String, message: String, handler: ((UIAlertAction) -> Void)? = nil) {
        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: handler))
        present(alert, animated: true, completion: nil)
    }
}
```

But since almost all our view controllers need alerts, we just extend UIViewController.

Reply:
Cool guys! thanks for commenting your thoughts :D

@cardoso that's true, in this case, it's better to just extend UIViewController.